### PR TITLE
[PPD-165] separazione ricevute

### PR DESCRIPTION
**Pull request by alessio-acitelli**
#### List of Changes

- added service filter to getOrganizationReceipts
- handled 404 error [PPD-180]

#### Motivation and Context

- separate the receipts to be sent to the PDA for the "canone unico"
- do not raise blocking error in case of 404 in the call to GPD

#### How Has This Been Tested?

JUnit and Integration test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

[PPD-180]: https://pagopa.atlassian.net/browse/PPD-180?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by alessio-acitelli** [comment]
create alerts on the log for 404 error management:
log.error("[getGPDCheckedReceiptsList] Non-blocking error: "
						+ "get not found exception in the recovery of payment options", e);

File: payments/src/main/java/it/gov/pagopa/payments/controller/receipt/IPaymentsController.java [hunk 1]
```diff
@@ -50,8 +50,6 @@ ResponseEntity<String> getReceiptByIUV(
     @ApiResponses(value = {
             @ApiResponse(responseCode = "200", description = "Obtained all organization payment positions.", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ReceiptsInfo.class))),
             @ApiResponse(responseCode = "401", description = "Wrong or missing function key.", content = @Content(schema = @Schema())),
-            @ApiResponse(responseCode = "404", description = "No receipts found.", content = @Content(schema = @Schema(implementation = ProblemJson.class))),
```
Comment (pasqualespica) [inline payments/src/main/java/it/gov/pagopa/payments/controller/receipt/IPaymentsController.java:53]: why removed 404 ?

File: payments/src/main/java/it/gov/pagopa/payments/service/PaymentsService.java [hunk 2]
```diff
@@ -114,10 +117,16 @@ public PaymentsResult<ReceiptEntity> getOrganizationReceipts(@Positive Integer l
...
 
 			String filter = TableQuery.generateFilterCondition(PARTITION_KEY_FIELD, TableQuery.QueryComparisons.EQUAL, organizationFiscalCode);
+			
 			if (null != debtor) {
 				String debtorFilter       = TableQuery.generateFilterCondition(DEBTOR_FIELD, TableQuery.QueryComparisons.EQUAL, debtor);
 				filter = TableQuery.combineFilters(filter, Operators.AND, debtorFilter);
 			}
+			
+			if (null != service) {
+				String serviceFilter = PaymentsService.getStartsWithFilter(ROW_KEY_FIELD, service);
```
Comment (pasqualespica) [inline payments/src/main/java/it/gov/pagopa/payments/service/PaymentsService.java:127]: rename `serviceFilter` with `segretationCodeFilter` to rember us that is a special case.
in the near future ( i hope ) we should service in more appropriately

File: payments/src/main/java/it/gov/pagopa/payments/service/PaymentsService.java [hunk 3]
```diff
@@ -130,15 +139,57 @@ public PaymentsResult<ReceiptEntity> getOrganizationReceipts(@Positive Integer l
...
+	public PaymentsResult<ReceiptEntity> getGPDCheckedReceiptsList(CloudTable table, PaymentsResult<ReceiptEntity> result){
+		// for all the receipts in the azure table, only those that have been already PAID status or are in PAID status on GPD are returned
+		List<ReceiptEntity> checkedReceipts = new ArrayList<>();
+		for (ReceiptEntity re: result.getResults()) {
+			try {
+				this.checkGPDDebtPosStatus(table, re);
+				checkedReceipts.add(re);
+			} 
+			catch (FeignException.NotFound e ) {
+				log.error("[getGPDCheckedReceiptsList] Non-blocking error: "
```
Comment (pasqualespica) [inline payments/src/main/java/it/gov/pagopa/payments/service/PaymentsService.java:176]: ok you'd try to open pr on https://github.com/pagopa/pagopa-infra to create something alarm like [that](https://github.com/pagopa/pagopa-infra/blob/d9106f6a19620f4af7f8cd7ac908f0247e18bde5/src/canoneunico.tf#L273)

File: payments/src/main/java/it/gov/pagopa/payments/service/PaymentsService.java [hunk 4]
```diff
@@ -176,53 +227,27 @@ private PaymentsResult<ReceiptEntity> getSegmentedReceipts(Integer limit, Intege
 
 	private PaymentsResult<ReceiptEntity> getNotSegmentedReceipts(CloudTable table, String filter) {
 		 final int FIRST_PAGE_NUMBER = 0;
-		 String[] columns = new String[]{PARTITION_KEY_FIELD, ROW_KEY_FIELD, DEBTOR_FIELD};
+		 String[] columns = new String[]{PARTITION_KEY_FIELD, ROW_KEY_FIELD, DEBTOR_FIELD, STATUS_FIELD};
```
Comment (pasqualespica) [inline payments/src/main/java/it/gov/pagopa/payments/service/PaymentsService.java:230]: why not a class `private static final String[] columns` both for segmentation or not. ?

**Review by pasqualespica — CHANGES_REQUESTED** [review]

File: payments/src/main/java/it/gov/pagopa/payments/controller/receipt/IPaymentsController.java (hunk 1 above)
Comment (alessio-acitelli) [reply payments/src/main/java/it/gov/pagopa/payments/controller/receipt/IPaymentsController.java:53]: As agreed we no longer return that kind of error externally. In the worst case scenario, the client will receive an empty list

File: payments/src/main/java/it/gov/pagopa/payments/service/PaymentsService.java (hunk 2 above)
Comment (alessio-acitelli) [reply payments/src/main/java/it/gov/pagopa/payments/service/PaymentsService.java:127]: Are you sure? In my opinion the implementation should be aseptic regarding the momentary meaning of a data that is passed in input

File: payments/src/main/java/it/gov/pagopa/payments/service/PaymentsService.java (hunk 4 above)
Comment (alessio-acitelli) [reply payments/src/main/java/it/gov/pagopa/payments/service/PaymentsService.java:230]: Yes, you are right, it is a useless duplication

**Review by pasqualespica — APPROVED** [review]

**Pull request merged**